docs/datastorage/attachments.md

```diff
@@ -0,0 +1,94 @@
+# Data Attachments
+
+The data attachment system allows mods to attach and store additional data on block entities, chunks, entities, and item stacks.
+
+_To store additional level data, you can use [SavedData](saveddata)._
+
+## Creating an attachment type
+
+To use the system, you need to register an `AttachmentType`.
+The attachment type contains:
+- a default value supplier to create the instance when the data is first accessed, or to compare stacks that have the data and stacks that don't have it;
+- an optional serializer if the attachment should be persisted;
+- additional configuration options for the attachment, for example the `copyOnDeath` flag.
+
+:::warning
+If you don't provide a serializer, the attachment will not be saved to disk.
+(This can be intended).
```

Comment:
This is a weird way of just saying 'If you don't want your attachment to persist its data, do not provide a serializer.'